src/chat_commands.c

```diff
@@ -367,29 +373,49 @@ void cmd_sendfile(WINDOW *window, ToxWindow *self, Tox *m, int argc, char (*argv
 
 on_send_error:
 
+    fclose(file_to_send);
+
     switch (err) {
-        case TOX_ERR_FILE_SEND_FRIEND_NOT_FOUND:
+        case TOX_ERR_FILE_SEND_FRIEND_NOT_FOUND: {
             errmsg = "File transfer failed: Invalid friend.";
             break;
+        }
 
-        case TOX_ERR_FILE_SEND_FRIEND_NOT_CONNECTED:
-            errmsg = "File transfer failed: Friend is offline.";
-            break;
+        case TOX_ERR_FILE_SEND_FRIEND_NOT_CONNECTED: {
+
+            int queue_idx = file_send_queue_add(self->num, path, path_len);
+
+            char msg[MAX_STR_SIZE];
+
+            if (queue_idx >= 0) {
+                snprintf(msg, sizeof(msg), "File transfer queued. Type \"/cancel out %d\" to cancel.", queue_idx);
+            } else if (queue_idx == -2) {
+                snprintf(msg, sizeof(msg), "File transfer queue is full.");
+            } else {
+                snprintf(msg, sizeof(msg), "Invalid file name.");
```

Comment:
Invalid because it's too long or empty. Maybe we can tell the user about this, possibly even distinguishing the cases.